src/Runner.Common/JobServerQueue.cs

```diff
@@ -744,7 +744,7 @@ private List<TimelineRecord> MergeTimelineRecords(List<TimelineRecord> timelineR
             foreach (var record in mergedRecords)
             {
                 Trace.Verbose($"    Record: t={record.RecordType}, n={record.Name}, s={record.State}, st={record.StartTime}, {record.PercentComplete}%, ft={record.FinishTime}, r={record.Result}: {record.CurrentOperation}");
-                if (record.Issues != null && record.Issues.Count > 0)
+                if (record.Issues != null)
                 {
                     foreach (var issue in record.Issues)
```

Comment:
`foreach` will just be a no-op if the `Count` is 0.  No need to check it explicitly.

Reply:
it's no-op... but you are wasting CPU cycles to do an extra `GetEnumerator()` and `MoveNext()`. 😄

Reply:
which is probably about equal to a `get_Issues`, a `get_Count` and a `jmp` evaluation ?

Reply:
I prefer to reduce changes in PR to only the required things to keep the review focus (especially if the change contains more files)

I am fine with the refactor if you like. 😄